examples/angular/auto-refetching/src/app/app.config.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  HTTP_INTERCEPTORS,
+  HttpClientModule,
+  provideHttpClient,
+  withFetch,
+} from '@angular/common/http'
+import {
+  QueryClient,
+  provideTanStackQuery,
+  withDevtools,
+} from '@tanstack/angular-query-experimental'
+import { importProvidersFrom } from '@angular/core'
+import { MockApiInterceptor } from './interceptor/mock-api.interceptor'
+import type { ApplicationConfig } from '@angular/core'
+
+export const appConfig: ApplicationConfig = {
+  providers: [
+    provideHttpClient(withFetch()),
+    provideTanStackQuery(
+      new QueryClient({
+        defaultOptions: {
+          queries: {
+            gcTime: 1000 * 60 * 60 * 24, // 24 hours
+          },
+        },
+      }),
+      withDevtools(),
+    ),
+    // Registers the MockApiInterceptor for simulating API calls
+    {
```

Comment:
Can you use a functional interceptor and `withInterceptors` instead? See the [rxjs example](https://github.com/TanStack/query/blob/b7bad3dfda38404bbdebd82f921dea94b9744220/examples/angular/rxjs/src/app/app.config.ts#L18)